### PR TITLE
Add fix to browser history after navigating to exploration page with no pipeline

### PR DESCRIPTION
**Pull request by nandhinibsn**
Resolve #

## Checklist:

You should check all boxes before the PR is ready. If a box does not apply, check it to acknowledge it.

* [x] **ISSUE NUMBER.** You linked the issue number (Ex: Resolve #XXX).
* [x] **PRE-COMMIT.** You ran pre-commit on all commits, or else, you
  ran `pre-commit run --all-files` at the end.
* [x] **USER CHANGES.** The changes are added to CHANGELOG.md and the documentation, if they impact
  our users.
* [x] **DEV CHANGES.**
    * Update the documentation if this PR changes how to develop/launch on the app.
    * Update the `README` files and our wiki for any big design decisions, if relevant.
    * Add unit tests, docstrings, typing and comments for complex sections.

### CODE REVIEW

**Comment by nandhinibsn** [comment]
There was a console error which occurred when we navigate to the exploration page with no pipeline.
Screen Shot 2023-02-23 at 5 17 01 PM

I simplified the solution to navigating on the Dashboard page instead of the Exploration page. I believe it's more of a simplified way. Let me know your thoughts @JosephMarinier

File: webapp/src/pages/Dashboard.tsx [hunk 1]
```diff
@@ -71,7 +71,9 @@ const Dashboard = () => {
           color="secondary"
           variant="contained"
           component={Link}
-          to={`/${jobId}/dataset_splits/${firstAvailableDatasetSplit}/prediction_overview${searchString}`}
+          to={`/${jobId}/dataset_splits/${firstAvailableDatasetSplit}/${
+            !isPipelineSelected(pipeline) ? `utterances` : `prediction_overview`
```
Comment (JosephMarinier) [inline webapp/src/pages/Dashboard.tsx:75]: I find the condition is easier to read if you remove the `!` and flip. Also, regular strings would be goof enough here.
```suggestion
            isPipelineSelected(pipeline) ? "prediction_overview" : "utterances"
```

**Comment by nandhinibsn** [comment]
Hey @JosephMarinier, the solution added to Dashboard has been reverted as it resolved the issue with tracking history. Still, it did not switch tabs from prediction overview to utterance table if no pipeline was selected from the exploration page.

Additionally, I understood why the console error was caused by reading this [blog](https://www.appsloveworld.com/reactjs/100/18/cannot-update-during-an-existing-state-transition-such-as-within-render-rend). 

To solve the bug, I added useEffect to handle the pipeline change from the exploration page to solve the issue with tracking history while the console error is not occurring anymore. Let me know what you think

**Comment by JosephMarinier** [comment]
Excellent! 👍 
As you said, the `React.useEffect` is perfect for if you select _No pipeline_ when already on the Exploration Space. However, when navigating from the Dashboard to the Exploration Space, it now works (👍) but it adds an intermediary state in the history, on the invalid Prediction Overview tab. So, I think the two solutions could be combined. You could also add your initial solution to the `Link to=` on the Dashboard, to polish the navigation. It also has the advantage of showing the correct destination when hovering the button.

File: webapp/src/pages/Exploration.tsx [hunk 2]
```diff
@@ -67,9 +67,11 @@ const Exploration = () => {
...
   };
 
-  if (!isPipelineSelected(pipeline) && mainView !== "utterances") {
-    setMainView("utterances");
-  }
+  React.useEffect(() => {
+    if (!isPipelineSelected(pipeline) && mainView !== "utterances") {
+      setMainView("utterances");
+    }
+  }, [pipeline]);
```
Comment (JosephMarinier) [inline webapp/src/pages/Exploration.tsx:74]: From the terminal
> React Hook React.useEffect has missing dependencies: 'mainView' and 'setMainView'.

File: webapp/src/pages/Dashboard.tsx (hunk 1 above)
Comment (nandhinibsn) [reply webapp/src/pages/Dashboard.tsx]: Sure. done

File: webapp/src/pages/Exploration.tsx (hunk 2 above)
Comment (nandhinibsn) [reply webapp/src/pages/Exploration.tsx:74]: I added all the necessary dependencies.

**Comment by nandhinibsn** [comment]
> Excellent! 👍 As you said, the `React.useEffect` is perfect for if you select _No pipeline_ when already on the Exploration Space. However, when navigating from the Dashboard to the Exploration Space, it now works (👍) but it adds an intermediary state in the history, on the invalid Prediction Overview tab. So, I think the two solutions could be combined. You could also add your initial solution to the `Link to=` on the Dashboard, to polish the navigation. It also has the advantage of showing the correct destination when hovering the button.

Okay, Make sense. I will add the condition back to the Dashboard page.

File: webapp/src/pages/Exploration.tsx (hunk 2 above)
Comment (JosephMarinier) [reply webapp/src/pages/Exploration.tsx]: You added only one of the two dependencies.

Comment (nandhinibsn) [reply webapp/src/pages/Exploration.tsx]: Okay, I was not aware adding a function to dependencies is acceptable. I added the dependencies. Please check my recent commit.

**Review by JosephMarinier — APPROVED** [review]
🚢 🇮🇹

**Pull request merged**